src/psyclone/tests/psyir/transformations/extract_test.py

```diff
@@ -600,20 +540,34 @@ def test_extract_kernel_and_builtin_dynamo0p3(tmpdir):
 
     schedule, _ = etrans.apply(schedule.children[1:3])
     code = str(psy.gen)
-    output = (
-        "      ! ExtractStart\n"
-        "      ! CALL write_extract_arguments(argument_list)\n"
-        "      !\n"
-        "      DO df=1,undf_any_space_1_f2\n"
-        "        f2_proxy%data(df) = 0.0\n"
-        "      END DO\n"
-        "      DO cell=1,f3_proxy%vspace%get_ncell()\n"
-        "        !\n"
-        "        CALL testkern_code_w2_only(nlayers, f3_proxy%data, "
-        "f2_proxy%data, ndf_w2, undf_w2, map_w2(:,cell))\n"
-        "      END DO\n"
-        "      !\n"
-        "      ! ExtractEnd\n")
+    output = """
+      ! ExtractStart
+      !
+      CALL psy_data%PreStart("single_invoke_builtin_then_kernel_psy", """ \
+      """"invoke_0:r0", 1, 3)
+      CALL psy_data%PreDeclareVariable("f2", f2)
+      CALL psy_data%PreDeclareVariable("cell_post", cell)
+      CALL psy_data%PreDeclareVariable("df_post", df)
+      CALL psy_data%PreDeclareVariable("f3_post", f3)
+      CALL psy_data%PreEndDeclaration
+      CALL psy_data%ProvideVariable("f2", f2)
+      CALL psy_data%PreEnd
+      DO df=1,undf_any_space_1_f2
+        f2_proxy%data(df) = 0.0
+      END DO
+      DO cell=1,f3_proxy%vspace%get_ncell()
+        !
+        CALL testkern_code_w2_only(nlayers, f3_proxy%data, """ + \
+        """f2_proxy%data, ndf_w2, undf_w2, map_w2(:,cell))
+      END DO
+      CALL psy_data%PostStart
+      CALL psy_data%ProvideVariable("cell_post", cell)
+      CALL psy_data%ProvideVariable("df_post", df)
+      CALL psy_data%ProvideVariable("f3_post", f3)
```

Comment:
Include code that should be doing `ProvideVariable` for f2_post and make this xfail please.

Reply:
Done.